### PR TITLE
Kog classifier

### PR DESCRIPTION
**Pull request by laurenssam**

### CODE REVIEW

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    "- De bezwaarmaker geeft aan contact te hebben gehad met de gemeente of een "
+    "melding te hebben gemaakt.\n"
+    "- De dader is minderjarig (tot en met 16 jaar)\n"
+    "- De bezwaarmaker schuift de schuld af op een ander (bijv. huisgenoot, kind of "
+    "buren), zonder zelf enige verantwoordelijkheid te erkennen.\n"
+    "- De bezwaarmaker stuurt bewijzen mee die aantonen dat hij/zij het niet kan zijn "
+    "geweest (bijvoorbeeld vliegtickets).\n"
+    "- De bezwaarmaker zegt het afval niet te herkennen of vermoedt dat iemand anders "
+    "het met zijn/haar naam erin heeft weggegooid.\n\n"
+    "Voorbeelden:\n"
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py:80]: Maybe a bit of a small thing again, but I would personally split the prompt into a template with placeholders to format. E.g.
have a separate list of examples which you neatly format into {EXAMPLES} somewhere there in the main prompt template. 

This would 
1) make the prompt easier to read (it would highlight the structure better) - like 

    f"{INTRO}"
    "De redenen zijn:"
    f"{REDEN}"
     ....

The other thing is that if you nicely format the voorbeelden and that stuff, it's easier to adjust the prompt - e.g. have a different format for 
"Bezwaar: ...  "
"Uitkomst": .... "
without changing it explicitly at 10 places :P

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    "Uitkomst: niet-KOG - De bezwaarmaker ontkent betrokkenheid bij de overtreding. "
+    "Hierom is het niet-KOG.\n"
+    "Voorbeeld 3 (niet-KOG)  \n"
+    'Bezwaar: "Ik wil bewijs zien, ik snap niet waarom ik deze boete krijg."\n'
+    "Uitkomst: niet-KOG - Er is onbegrip en geen erkenning van de overtreding. Hierom "
+    "is het niet-KOG.\n"
+    "Voorbeeld 4 (KOG)  \n"
+    'Bezwaar: "Ik was op vakantie en zette mijn afval eerder buiten."  \n'
+    "Uitkomst: KOG - De bezwaarmaker erkent de overtreding. Hierom is het KOG.\n"
+    "Voorbeeld 5 (niet-KOG)  \n"
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py:97]: Shouldn't the examples be aligned with the expected format? I see no final "Uitkomst" at the end of our prompt

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    "Uitkomst: niet-KOG - De bezwaarmaker ontkent de overtreding en schuift de schuld "
+    "af.\n"
+    "Voorbeeld 12 (KOG)\n"
+    'Bezwaar: "Ik heb mijn afval netjes weggegooid in de container, alleen is het dan '
+    'waarschijnlijk eruit gevallen."\n'
+    "Uitkomst: KOG - De bezwaarmaker geeft aan dat het afval waarschijnlijk eruit is "
+    "gevallen, wat geen geldige reden is en het bezwaar hierdoor KOG is.\n"
+    "Wees consistent: geef per bezwaar exact één regel terug met het label 'KOG –' of "
+    "'niet-KOG –' en een korte reden.\n"
+    "Bezwaarschrift:\n"
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py:139]: yeah, about the voorbelden above - if few shot examples are "Bezwaar: ..." I would also keep this "Bezwaar: ... \n Uitkomst: ..." (not "Bezwaarshift: ...")

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+        self.data = self._load_data()
+
+    def _load_data(self):
+        """Load data from local file."""
+        df = pd.read_excel(self.data_path)
+        return df.to_dict("records")
+
+    def _get_hashing_data_for_sampling(self):
+        """Get data for consistent sampling."""
+        # Usually return a list of unique identifiers
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py:177]: Btw the hashing trick uses strings, I usually pass the raw docs themselves for shuffling, but ids are also fine... it just mmh3 just turns them into strings first anyway :D

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+        return [entry["id"] for entry in self.data]
+
+    def _run_task(self, llm, n_samples=0) -> List[RunItem]:
+        """Run the classification task using the provided LLM."""
+        if self.data is None:
+            raise ValueError("Benchmark data is not loaded.")
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py]: why raise instead of just calling the loading function? To avoid loading at inference time for the models, or? :/

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+        Returns: 1 (KOG), 0 (Not-KOG), or "invalid"
+        """
+        if not response:
+            return "INVALID"
+
+        # Clean and prepare response
+        response = str(response).strip()
+
+        # Check if it matches expected patterns (case-insensitive)
+        if re.match(r"^KOG\s*[-–—]\s*.+", response, re.IGNORECASE):
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py:225]: Your examples from the voorbeeld have "Uitkomst: ..." 
so if we don't add it to the main prompt or adjust the examples, then I would say make sure it's being processed properly and ignored (so KOG can be at start of string or after Uitkomst:...)

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+        }
+
+        # Calculate precision, recall, F1 only on valid predictions
+        if n_valid > 0:
+            # Extract targets and predictions for valid items only
+            targets = [int(item.target) for item in valid_items]
+            predictions = [int(item.predicted) for item in valid_items]
+
+            # Calculate confusion matrix components
+            tp = sum(
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py:271]: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html ? :D

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+
+        return BenchmarkEvaluation(
+            metrics=metrics,
+            total_samples=len(run_output),
+        )
+
+    def _get_own_metadata(self):
+        """Get benchmark-specific metadata."""
+        metadata = {
+            "task_type": "classification",
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py:328]: Can we maybe also log the prompt?
I think it would be good in case we want to experiment with a few versions or update it eventually!!

File: llm_eval/benchmarks/use_cases/kog_classifier.py [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+            "language": "dutch",
+        }
+        return metadata
+
+    def _load_prompt(self):
```
Comment (igornishka) [inline llm_eval/benchmarks/use_cases/kog_classifier.py]: Unused?

**Review by igornishka — APPROVED** [review]
